pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -321,88 +321,33 @@ protected void removeColumnV1Indices(String column)
   }
 
   /**
-   * Right now the text index is supported on RAW (non-dictionary encoded)
-   * single-value STRING columns. Eventually we will relax the constraints
-   * step by step.
-   * For example, later on user should be able to create text index on
-   * a dictionary encoded STRING column that also has native Pinot's inverted
-   * index. We can also support it on BYTE columns later.
+   * Right now the text index is supported on RAW and dictionary encoded
+   * single-value STRING columns. Later we can add support for text index
+   * on multi-value columns and BYTE type columns
    * @param column column name
-   * @param indexLoadingConfig index loading config
    * @param fieldSpec field spec
    */
-  private void checkUnsupportedOperationsForTextIndex(String column, IndexLoadingConfig indexLoadingConfig,
-      FieldSpec fieldSpec) {
-    if (!indexLoadingConfig.getNoDictionaryColumns().contains(column)) {
-      throw new UnsupportedOperationException(
-          "Text index is currently not supported on dictionary encoded column: " + column);
-    }
-
-    Set<String> sortedColumns = new HashSet<>(indexLoadingConfig.getSortedColumns());
-    if (sortedColumns.contains(column)) {
-      // since Pinot's current implementation doesn't support raw sorted columns,
-      // we need to check for this too
-      throw new UnsupportedOperationException("Text index is currently not supported on sorted column: " + column);
-    }
-
+  private void checkUnsupportedOperationsForTextIndex(String column, FieldSpec fieldSpec) {
```

Comment:
Let's remove this check as it will be checked in `TextIndexHandler`. Try not to couple different modules together.